common/src/main/java/io/pravega/common/health/processor/impl/HealthRequestProcessorImpl.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.health.processor.impl;
+
+import com.google.common.base.Strings;
+import io.pravega.common.health.HealthReporter;
+import io.pravega.common.health.NoSuchHealthCommand;
+import io.pravega.common.health.NoSuchHealthProcessor;
+import io.pravega.common.health.processor.HealthRequestProcessor;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class HealthRequestProcessorImpl extends HealthRequestProcessor {
+    private final ConcurrentHashMap<String, HealthReporter> reporterMap = new ConcurrentHashMap<>();
+
+    public HealthRequestProcessorImpl() {
+        super();
+    }
+
+    @Override
+    public void registerHealthProcessor(String id, HealthReporter reporter) {
+        reporterMap.putIfAbsent(id, reporter);
+    }
+
+    @Override
+    public final void processHealthRequest(OutputStream writer, String target, String cmd) throws IOException {
+        DataOutputStream opStr = new DataOutputStream(writer);
+        if (!Strings.isNullOrEmpty(target)) {
+            HealthReporter reporter = reporterMap.get(target);
+            if (reporter != null) {
+                Consumer<DataOutputStream> handler = reporter.getHandlers().get(cmd);
+                if (handler != null) {
+                    handler.accept(new DataOutputStream(writer));
+                } else {
+                    throw new NoSuchHealthCommand(cmd);
+                }
+            } else {
+                throw new NoSuchHealthProcessor(target);
+            }
+        } else {
+            switch (cmd) {
+                case "tgts":
```

Comment:
This is a bit confusing, we are getting handlers out of the health reporter, and currently, the only handler available is the ruok one. This process method is handling a command in line, though. Specifically, this is handling `tgts` directly rather than do it via a handler, which seems to be the model you are trying to promote.

Reply:
Made `HealthProcessor` also as a `HealthReporter`. Doing this the right way now.